src/test/scala/com/gu/IntegrationTests.scala

```diff
@@ -133,19 +156,21 @@ class IntegrationTests extends FlatSpec with Matchers with Http with Eventually
       s"Test attachment $timestamp",
       "This is a test attachment",
       "Fallback text"
-    ).withPretext(s"Test pretext $timestamp").withColor("#FEFEFD").withThumbUrl("https://image.freepik.com/free-icon/thumb-up-sign_318-63754.jpg")
-
-    val response = new SlackIncomingWebHook(webhookUrl).send(Payload(s"Test post - optional parameter attachment test $timestamp").withAttachment(attachment))
-    response.responseCode should be (200)
-
-    val channelApi = SlackApiChannels(generalChannel)
-
-    List(
-      attachment.pretext.get -> channelApi.firstAttachmentPretext,
-      attachment.color.get.replace("#", "") -> channelApi.firstAttachmentColor,
-      attachment.thumb_url.get -> channelApi.firstAttachmentThumbUrl
-    ).foreach { e =>
-      e._1 should be (e._2)
+    )
+      .withPretext(s"Test pretext $timestamp")
+      .withColor("#FEFEFD")
+      .withThumbUrl("https://image.freepik.com/free-icon/thumb-up-sign_318-63754.jpg")
```

Comment:
I don't love this mix of constructor and builder styles. You could just set all the properties when you create the instance?

```scala
Attachment(
s"Test attachment $timestamp",
"This is a test attachment",
"Fallback text",
pretext = Some(s"Test pretext $timestamp"),
color = Some("#FEFEFD)",
thumbUrl = Some("https://image.freepik.com/free-icon/thumb-up-sign_318-63754.jpg")
)
```